cmd/monaco/download/download_command.go

```diff
@@ -16,140 +16,115 @@ package download
-		RunE: func(cmd *cobra.Command, args []string) error {
-			m := args[0]
-			specificEnvironment := args[1]
-			options := manifestDownloadOptions{
-				manifestFile:            m,
-				specificEnvironmentName: specificEnvironment,
-				downloadCommandOptions: downloadCommandOptions{
-					downloadCommandOptionsShared: downloadCommandOptionsShared{
-						projectName:    project,
-						outputFolder:   outputFolder,
-						forceOverwrite: forceOverwrite,
-					},
-					specificAPIs:    specificApis,
-					specificSchemas: specificSettings,
-					onlyAPIs:        onlyAPIs,
-					onlySettings:    onlySettings,
-				},
-			}
+	downloadCmd = &cobra.Command{
+		Use:   "download --manifest <manifest.yaml> --token <API_TOKEN_ENV_VAR_NAME>",
+		Short: "Download configuration from Dynatrace",
+		Long: `Download configuration from Dynatrace
 
-			return command.DownloadConfigsBasedOnManifest(fs, options)
-		},
-	}
+Either downloading based on an existing manifest, or define an URL pointing to an environment to download configuration from.`,
 
-	directDownloadCmd := &cobra.Command{
-		Use:     "direct [URL] [TOKEN_NAME]",
-		Aliases: []string{"d"},
-		Short:   "Download configuration from a Dynatrace environment specified on the command line",
-		Example: `monaco download direct https://environment.live.dynatrace.com API_TOKEN_ENV_VAR_NAME`,
-		Args: func(cmd *cobra.Command, args []string) error {
-			if len(args) != 2 || args[0] == "" || args[1] == "" {
-				return fmt.Errorf(`url and token have to be provided as positional argument`)
+		Example: `- monaco download --manifest manifest.yaml --token API_TOKEN_ENV_VAR_NAME
+- monaco download --url https://environment.live.dynatrace.com --token API_TOKEN_ENV_VAR_NAME`,
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if url == "" && (oAuthClientID != "" || oAuthClientSecret != "") {
+				return errors.New("if any flags from the group [oauth-client-id oauth-client-secret] is set [url] flag must also be set")
 			}
 			return nil
 		},
-		ValidArgsFunction: completion.DownloadDirectCompletion,
-		PreRun: func(cmd *cobra.Command, args []string) {
-			serverVersion, err := client.GetDynatraceVersion(client.NewTokenAuthClient(os.Getenv(args[1])), args[0])
-			if err != nil {
-				log.Error("Unable to determine server version %q: %w", args[0], err)
-				return
-			}
-			if serverVersion.SmallerThan(version.Version{Major: 1, Minor: 262}) {
-				logUploadToSameEnvironmentWarning()
-			}
-		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			url := args[0]
-			tokenEnvVar := args[1]
-			options := directDownloadOptions{
-				environmentUrl: url,
-				envVarName:     tokenEnvVar,
-				downloadCommandOptions: downloadCommandOptions{
-					downloadCommandOptionsShared: downloadCommandOptionsShared{
-						projectName:    project,
-						outputFolder:   outputFolder,
-						forceOverwrite: forceOverwrite,
+			if url != "" {
+				manifestFile = ""
```

Comment:
it's not necessary to do this is it? the content isn't used anyway in this logical branch